dependencies/influxdb/http_provider_test.go

```diff
@@ -0,0 +1,32 @@
+package influxdb
+
+import (
+	"context"
+	"testing"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/dependencies/http"
+	fluxurl "github.com/influxdata/flux/dependencies/url"
+	"github.com/influxdata/flux/dependency"
+)
+
+func TestPrivateClient(t *testing.T) {
+	h := HttpProvider{
+		DefaultConfig: Config{
+			Host: "http://myhost.com:8085",
+		},
+	}
+	deps := flux.Deps{Deps: flux.WrappedDeps{
+		HTTPClient:   http.NewDefaultClient(fluxurl.PrivateIPValidator{}),
+		URLValidator: fluxurl.PrivateIPValidator{},
+	}}
+	ctx, _ := dependency.Inject(context.Background(), deps)
+	c, err := h.clientFor(ctx, Config{})
+	if err != nil {
+		t.Error(err)
+	}
+	client := c.Client
+	if _, ok := client.(*http.PrivateClient); !ok {
+		t.Error("reader does not use private client")
+	}
+}
```

Comment:
The `HTTPClient` field in `Deps` is an interface, so you can pass anything you want in there.

What if you passed a mock client into the `Deps` that looks something like this
```go
import nethttp "net/http"
import "github.com/influxdata/flux/dependencies/http"

type struct mockClient {}

var _ http.Client = mockClient{}

func (c mockClient) Do(req *nethttp.Request) (*nethttp.Response, error) {
  return <some error>
}
```

Then create an instance of `Deps` with the mock client. You should be able to show that when the client returns an error you get the `an internal error...` message if there was no URL supplied.

This way both `PrivateClient` and `NewPrivateClient` don't need to be exported, but you can still test the new logic you added.

Reply:
I was able to make `PrivateClient` private again using your suggestions, but I don't see how to make `NewPrivateClient` private, since it needs to be imported to a separate package in order for `Deps.PrivateHTTPClient()` to work.

https://github.com/influxdata/flux/blob/648369db57a595e9bcaab076c9f5815c373b4c37/dependencies.go#L55-L62